content/flux/v0.x/data-types/basic/integer.md

```diff
@@ -2,11 +2,198 @@
 title: Work with integers
 list_title: Integer
 description: >
-  ...
+  Learn how to work with integer types in Flux.
 menu:
   flux_0_x:
     name: Integer
     parent: Basic types
 weight: 202
 flux/v0.x/tags: ["basic types", "numeric types", "data types"]
----
\ No newline at end of file
+related:
+  - /flux/v0.x/stdlib/universe/int/
+  - /flux/v0.x/stdlib/universe/toint/
+---
+
+An **integer** type represents a signed 64-bit integer.
+
+**Type name**: `int`  
+**Min value**: `-9223372036854775808`  
+**Max value**: `9223372036854775807`
+
+###### On this page:
+- [Integer syntax](#integer-syntax)
+- [Convert data types to integers](#convert-data-types-to-integers)
+- [Operate on integers](#operate-on-integers)
+
+## Integer syntax
+An integer literal contains one or more digits (0-9) optionally preceded by
+`-` to indicate a negative number.
+`-0` is equivalent to `0` and is not a negative number.
+
+```js
+0
+2
+1254
+-1254
+```
+
+## Convert data types to integers
+Use the [`int()` function](/flux/v0.x/stdlib/universe/int/) to convert
+the following [basic types](/flux/v0.x/data-types/basic/) to integers:
+
+- **string**: returns the integer equivalent of the numeric string (`[0-9]`)
+- **bool**: returns `1` for `true` or `0` for `false`
+- **duration**: returns to the number of nanoseconds in the duration
+- **time**: returns the equivalent [nanosecond epoch timestamp](/influxdb/cloud/reference/glossary/#unix-timestamp)
+- **float**: truncates the float value at the decimal
+- **uint**: returns the integer equivalent of the unsigned integer
+
+```js
+int(v: "123")
+// 123
+
+int(v: true)
+// Returns 1
+
+int(v: 1d3h24m)
+// Returns 98640000000000
+
+int(v: 2021-01-01T00:00:00Z)
+// Returns 1609459200000000000
+
+int(v: 12.54)
+// Returns 12
+```
+
+{{% note %}}
+#### Round float values before converting to integers
+When converting a [float](/flux/v0.x/data-types/basic/float/) value to an integer,
+Flux _truncates_ the float value at the decimal.
+To round float values to the nearest whole number:
```

Comment:
These aren't the same? 🤔

Reply:
No, truncated means just cutting off everything after the decimal. For example:

- `1.54` truncated at the decimal is `1`
- `1.54` rounded to the nearest whole number is `2`

Reply:
```suggestion
Being Flux _truncates_ the [float](/flux/v0.x/data-types/basic/float/) value at the decimal when converting to an integer, for example `1.54` to `1`, you may want to round float values to the nearest whole number `1.541 to `2` before converting. To do this:
```